### DISCUSSION TITLE
Failed to initialize any reasonable values

### DISCUSSION THREAD

**Discussion by marimaro** [hahnlab/CAFE5#132] [Q&A] [answered]
Hello,

I am trying to do an expansion/retraction analysis on a plant genome dataset. I have 10 different species (both monocots and eudicots with different ploidies) and I keep encountering this error (I have already removed a few families that had really high differences in species I was not so interested in).

I don't understand why this is happening, because if I am searching for expansions, this is exactly what I am looking for, right? Many genes in a family in a single species. 

Can you please help me in some way? I have used the script `clade_and_size_filter.py` to filter the dataset and tried different gamma categories values and the error persists.

Thank you in advance!

> 
> Families with largest size differentials:
> OG0000346: 27
> OG0000031: 26
> OG0000136: 26
> OG0000227: 25
> OG0000161: 25
> OG0000118: 25
> OG0000214: 25
> OG0000364: 25
> OG0000100: 25
> OG0000620: 24
> OG0000080: 24
> OG0000035: 24
> OG0000078: 24
> OG0000106: 23
> OG0001614: 23
> OG0000122: 23
> OG0001245: 23
> OG0000782: 23
> OG0000575: 23
> OG0000871: 23
> 
> You may want to try removing the top few families with the largest difference
> between the max and min counts and then re-run the analysis.
> 
> Failed to initialize any reasonable values
>

**Comment by sajjadasaf** [comment]
Hello,
I have got same problem. Anybody have solved  this issue?

**Comment by benfulton** [comment]
You may have better luck asking on the mailing list. https://groups.google.com/g/hahnlabcafe

**Comment by rsandean** [comment]
Hi, has anyone solved this as I am also experiencing the same problem :) Thank you!

**Comment by benfulton** [comment, marked as answer]
This error indicates that CAFE couldn't even find a starting point to begin running the optimizer. It could be that your tree is simply too big, and/or your families too variable, for CAFE to make sense of the data. As there is likely to be lots of conflict deep in the tree, an option would be to use sub-clades of the tree in independent runs. Dr. Hahn also says, "You could also try parsimony-based methods (e.g. “Count”), which would be guaranteed to give you something!"

**Comment by shitingyu083** [comment]
Hello,
I apologize for disturbing you. I encountered the following issue while using CAFE. The error message displayed on the screen is:
"You may want to try removing the top few families with the largest difference between the max and min counts and then re-run the analysis."
"Failed to initialize any reasonable values."
[errorwork.txt](https://github.com/user-attachments/files/17321266/errorwork.txt)

Previously, I saw a suggestion to reduce the -l value, so I attempted the following commands:
cafe5 -i test3.txt -t tree.txt -o out -c 16 -l 0.00001 -k 2 -p -a 0.5
and
cafe5 -i test3.txt -t tree.txt -o out -c 16 -l 0.0001 -k 2 -p,
but the error persists. After searching for similar error messages, I didn’t find any successful debugging experiences. I have attached my work log and input files for your reference.
If possible, could you please reply to help resolve the issue?
Thank you!
Best regards,

**Reply by homefucker** [reply]
Thankyou sir,After adding the -l parameter, my data started to run successfully. Regardless of the principle, I can finally go to sleep now. It's going to 5 am soon.
